packages/hmr-plugin/src/internal/common.ts

```diff
@@ -0,0 +1,27 @@
+import { NgModuleRef } from '@angular/core';
+import { NgxsBootstrapper, StateStream, Store } from '@ngxs/store';
+
+import { NGXS_HMR_SNAPSHOT_KEY, NgxsStoreSnapshot } from '../symbols';
+
+export function validateExistHmrStorage() {
+  const hmrStorageDoesNotExist = !(sessionStorage.getItem(NGXS_HMR_SNAPSHOT_KEY) || null);
+  if (hmrStorageDoesNotExist) {
+    sessionStorage.setItem(NGXS_HMR_SNAPSHOT_KEY, '{}');
+  }
+}
+
+export function getStateFromHmrStorage<S = NgxsStoreSnapshot>(): Partial<S> {
+  return JSON.parse(sessionStorage.getItem(NGXS_HMR_SNAPSHOT_KEY) || '{}');
+}
+
+export function setStateInHmrStorage<S = NgxsStoreSnapshot>(state: S): void {
+  return sessionStorage.setItem(NGXS_HMR_SNAPSHOT_KEY, JSON.stringify(state));
+}
+
+export function getStateStream<T>(ref: NgModuleRef<T>): StateStream | null {
+  return ref.injector.get(Store, null)['_stateStream'] || null;
```

Comment:
If injector's get returns `null` - it will throw `Cannot read property '_stateStream' of null`

```TS
const store = ref.injector.get<Store>(Store, null);
if (store) {
  return store['_stateStream'];
}

return null;
```